### DISCUSSION TITLE
creating a hook that's an observer

### DISCUSSION THREAD

**Discussion by llaenowyd** [mobxjs/mobx#2967] [Q&A]
Hi, googling this is kind of a trick since it just wants to show pages about `useObserver` - that is not my question.

So I've got my app running along with stuff like

```js
import React from 'react'
import {observer} from 'mobx-react-lite'

const Widget = () => {
  const store = useStore()
  
  return (
    <div><pre>{store.data}</pre></div>
  )
}

export default observer(Widget)
```

But now what I wanted was to create a hook that uses the store, like
```js
const customHook = () => {
  const store = useStore()
  
  return makeSomeFunctionsOnTheStore(store.data)
}

export default observer(customHook) // <-- this is apparently completely wrong
```

Wonder what's the best approach from here...  Thanks!

**Comment by mweststrate** [comment]
See https://mobx.js.org/react-integration.html using React Context. Users
of your hook will still need to apply `observer`.

**Reply by llaenowyd** [reply]
Right, I have a provider, [here it is](https://github.com/llaenowyd/next-task-management/blob/main/components/StoreProvider.tsx), so I think you mean, I don't need to decorate the hook as an observer, it just works as-is?

**Reply by llaenowyd** [reply]
👍 seems to work fine unless I'm missing something... thanks!
